perceval/backends/git.py

```diff
@@ -632,25 +660,46 @@ def log(self, from_date=None, encoding='utf-8'):
 
         :raises RepositoryError: when an error occurs fetching the log
         """
+
         cmd_log = ['git', 'log', '--raw', '--numstat', '--pretty=fuller',
                    '--decorate=full', '--all', '--reverse', '--topo-order',
                    '--parents', '-M', '-C', '-c', '--remotes=origin']
-
         if from_date:
             dt = from_date.strftime("%Y-%m-%d %H:%M:%S %z")
             cmd_log.append('--since=' + dt)
+        env = {'LANG' : 'C', 'PAGER' : ''}
 
-        gitlog = self._exec(cmd_log, cwd=self.dirpath,
-                            env={'LANG' : 'C', 'PAGER' : ''})
-        gitlog = gitlog.split(b'\n')
-
+        self.failed_message = None
```

Comment:
I think it makes more sense to move all this code to `_exec` method because it's generic and can be useful for other (future) commands.